src/main/java/org/fusesource/jansi/io/HtmlAnsiOutputStream.java

```diff
@@ -0,0 +1,176 @@
+ * @author <a href="https://github.com/Osiris-Team">Osiris Team</a>
+ * @author <a href="http://code.dblock.org">Daniel Doubrovkine</a>
+ */
+public class HtmlAnsiOutputStream extends AnsiOutputStream {
+
+    @Override
+    public void close() throws IOException {
+        closeAttributes();
+        super.close();
+    }
+
+    private static final String[] ANSI_COLOR_MAP = {"black", "red",
+            "green", "yellow", "blue", "magenta", "cyan", "white",};
+
+    private static final byte[] BYTES_QUOT = "&quot;".getBytes();
+    private static final byte[] BYTES_AMP = "&amp;".getBytes();
+    private static final byte[] BYTES_LT = "&lt;".getBytes();
+    private static final byte[] BYTES_GT = "&gt;".getBytes();
+
+    public HtmlAnsiOutputStream(OutputStream os) {
+        super(os, // With jdk8 the below could be simplified to:
+                // AnsiConsole.out()::getTerminalWidth
+                new WidthSupplier() {
+                    @Override
+                    public int getTerminalWidth() {
+                        return AnsiConsole.out().getTerminalWidth();
+                    }
+                },
+                AnsiConsole.out().getMode(),
+                null,
+                AnsiConsole.out().getType(),
+                AnsiConsole.out().getColors(),
+                Charset.defaultCharset(),
+                null,
+                null,
+                true);
+        AnsiToHtmlProcessor ansiToHtmlProcessor = new AnsiToHtmlProcessor(AnsiConsole.out());
+        ansiToHtmlProcessor.setHtmlAnsiOutputStream(this);
+        this.setAp(ansiToHtmlProcessor);
+        this.setProcessor(ansiToHtmlProcessor);
+    }
+
+    private final List<String> closingAttributes = new ArrayList<>();
+
+    private void write(String s) throws IOException {
+        super.out.write(s.getBytes());
+    }
+
+    private void writeAttribute(String s) throws IOException {
+        write("<" + s + ">");
+        closingAttributes.add(0, s.split(" ", 2)[0]);
+    }
+
+    private void closeAttributes() throws IOException {
+        for (String attr : closingAttributes) {
+            write("</" + attr + ">");
+        }
+        closingAttributes.clear();
+    }
+
+    public void write(int data) throws IOException {
+        switch (data) {
+            case 34: // "
+                out.write(BYTES_QUOT);
+                break;
+            case 38: // &
+                out.write(BYTES_AMP);
+                break;
+            case 60: // <
+                out.write(BYTES_LT);
+                break;
+            case 62: // >
+                out.write(BYTES_GT);
+                break;
+            default:
+                super.write(data);
+        }
+    }
+
+    public void writeLine(byte[] buf, int offset, int len) throws IOException {
+        write(buf, offset, len);
+        closeAttributes();
+    }
+
+    private static class AnsiToHtmlProcessor extends AnsiProcessor {
```

Comment:
Idk why this must be a static nested class...

Reply:
Agreed, especially if the HtmlAnsiOutputStream is passed and stored in a field.  It should simply be an inner class.